cs/test/core/TypeAliasTests.cs

```diff
@@ -284,6 +434,82 @@ static BlobAlias InitBlobAlias()
         }
     }
 
+    public class Box<T> : IEquatable<Box<T>>
+    {
+        public T Value { get; }
+
+        public Box()
+        {
+            Value = default(T);
+        }
+
+        public Box(T value)
+        {
+            Value = value;
+        }
+
+        public static implicit operator Box<T>(T value)
+        {
+            return new Box<T>(value);
+        }
+
+        public static bool operator ==(Box<T> left, Box<T> right)
```

Comment:
Check whether switching to a `struct` has the correct equality semantics and can get rid of a lot of this code.

Or, find some other dummy wrapper type elsewhere in the code base.